backend/capellacollab/projects/capellamodels/routes.py

```diff
@@ -114,6 +114,20 @@ def patch_capella_model(
     return crud.update_model(db, model, body.description, version, nature)
 
 
+@router.delete(
+    "/{model_slug}",
+    dependencies=[
+        Depends(ProjectRoleVerification(required_role=ProjectUserRole.MANAGER))
+    ],
+    tags=["Projects - Models"],
+)
+def patch_capella_model(
+    model: DatabaseCapellaModel = Depends(get_existing_capella_model),
+    db: Session = Depends(get_db),
+) -> DatabaseCapellaModel:
+    crud.delete_model(db, model)
```

Comment:
```suggestion
):
    if not (len(model.git_models) or len(model.t4c_models)):
        crud.delete_model(db, model)
```
Here I would add a backend validation (in addition to the frontend check). In case there are still git/t4c models attached, we could also throw an http exception.